### PR TITLE
chore: adding CEL for psp-host-network-ports policy

### PR DESCRIPTION
**Pull request by JaydipGabani**
**What this PR does / why we need it**:

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #541 

**Special notes for your reviewer**:

### CODE REVIEW

File: library/pod-security-policy/host-network-ports/template.yaml [hunk 1]
```diff
@@ -46,73 +46,111 @@ spec:
...
+            expression: |
+              (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                container.image in variables.exemptImageExplicit ||
+                variables.exemptImagePrefixes.exists(exemption, string(container.image).startsWith(exemption)))
+          - name: badContainers
+            expression: |
+              (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                !(container.image in variables.exemptImages) &&
+                (
+                  (has(variables.params.hostNetwork) && variables.params.hostNetwork ? (has(variables.anyObject.spec.hostNetwork) && variables.anyObject.spec.hostNetwork) : false) ||
```
Comment (maxsmythe) [inline library/pod-security-policy/host-network-ports/template.yaml:77]: `!variables.params.hostNetwork`?

The Rego reads like: "if hostNetwork parameter is false, make sure `spec.hostNetwork` is either unset or false.

File: artifacthub/library/pod-security-policy/host-network-ports/1.1.0/suite.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+metadata:
+  name: host-network-ports
+tests:
+- name: use-of-host-networking-ports-blocked
+  template: template.yaml
+  constraint: samples/psp-host-network-ports/constraint.yaml
```
Comment (maxsmythe) [inline artifacthub/library/pod-security-policy/host-network-ports/1.1.0/suite.yaml:8]: We probably should add a hostNetwork test to detect bugs like the one highlighted below.

File: library/pod-security-policy/host-network-ports/template.yaml (hunk 1 above)
Comment (JaydipGabani) [reply library/pod-security-policy/host-network-ports/template.yaml]: good catch! corrected the bug

File: artifacthub/library/pod-security-policy/host-network-ports/1.1.0/suite.yaml (hunk 2 above)
Comment (JaydipGabani) [reply artifacthub/library/pod-security-policy/host-network-ports/1.1.0/suite.yaml:8]: yeah, added a the tests to `suite.yaml` for this.

**Review by ritazh — APPROVED** [review]
LGTM

File: library/pod-security-policy/host-network-ports/template.yaml [hunk 3]
```diff
@@ -46,73 +46,111 @@ spec:
...
+            expression: |
+              (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                container.image in variables.exemptImageExplicit ||
+                variables.exemptImagePrefixes.exists(exemption, string(container.image).startsWith(exemption)))
+          - name: badContainers
+            expression: |
+              (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                !(container.image in variables.exemptImages) &&
+                (
+                  (has(variables.params.hostNetwork) && !variables.params.hostNetwork ? (has(variables.anyObject.spec.hostNetwork) && variables.anyObject.spec.hostNetwork) : false) ||
```
Comment (maxsmythe) [inline library/pod-security-policy/host-network-ports/template.yaml:77]: to be 1:1 with rego, if variables.params.hostNetwork is undefined, we should check ensure spec.hostNetwork is false (`not input.parameters.hotNetwork` is the Rego statement)

Basically: `!has(variables.params.hostNetwork) || !variables.params.hostNetwork)`

**Review by maxsmythe — CHANGES_REQUESTED** [review]

File: library/pod-security-policy/host-network-ports/template.yaml (hunk 3 above)
Comment (JaydipGabani) [reply library/pod-security-policy/host-network-ports/template.yaml]: Fixed

**Review by maxsmythe — APPROVED** [review]
LGTM

**Pull request merged**